datastore/src/main/java/org/jboss/pnc/datastore/repositories/RepositoryConfigurationRepositoryImpl.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2014 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.pnc.datastore.repositories;
+
+import org.jboss.pnc.datastore.repositories.internal.AbstractRepository;
+import org.jboss.pnc.datastore.repositories.internal.RepositoryConfigurationSpringRepository;
+import org.jboss.pnc.model.RepositoryConfiguration;
+import org.jboss.pnc.spi.datastore.repositories.RepositoryConfigurationRepository;
+
+import javax.ejb.Stateless;
+import javax.inject.Inject;
+
+/**
+ * @author Jakub Bartecek
+ */
+@Stateless
+public class RepositoryConfigurationRepositoryImpl extends AbstractRepository<RepositoryConfiguration, Integer> implements RepositoryConfigurationRepository {
+
+    /**
+     * @deprecated Created for CDI.
+     */
+    @Deprecated
+    public RepositoryConfigurationRepositoryImpl() {
+    }
```

Comment:
No it's not. If the injection point is using the interface, you don't have to have non-parametric constructor on the implementing class. For more info: http://docs.jboss.org/weld/reference/2.4.0.CR1/en-US/html/injection.html#_client_proxies